odoo/addons/base/views/res_users_views.xml

```diff
@@ -334,6 +337,17 @@
                 <field name="groups_id" position="after"/>
             </field>
         </record>
+
+        <record id="user_groups_tree_view" model="ir.ui.view">
+            <field name="name">res.users.tree.groups</field>
+            <field name="model">res.users</field>
+            <field name="inherit_id" ref="view_users_tree"/>
+            <field name="arch" type="xml">
+                <!-- dummy, will be modified by groups -->
+                <field name="groups_id" position="after"/>
```

Comment:
Seems strange, what is the purpose of this code ?

Reply:
The added field will be used and modified by the `_update_user_groups_tree_view(self)` to include the new portal users filter.

Reply:
Would really prefer using the share field, or if really you target portal (and not public), add a computed / searchable field for search view. Bref this is quite an usine à gaz that I think we could skip...

Reply:
Yes we want to target portal specifically, already tried a computed field, but it does not work :/

The issue here is that this field is created and manipulated programatically during runtime.

Reply:
Then maybe add a _is_portal field doing a compute / search on base.group_portal group being in user groups instead of using the reified computation that is quite noisy.

Reply:
In fact I was mistaken, the field we are adding here is called `user_type` and it's purpose is to show the user type of the user in the tree view (internal user, portal, public...), like we do in the form view, and that's why we need to create it dynamically as it's done for the form view.

For the search/filter field we are indeed using the share field.

Sorry for the back and forth.

Reply:
Actually the whole form view hack should stay localized. It is done to ease group management for access rights ... in our case I think we should define and use a real field for tree view. Otherwise code is really complicated although it should stay simple as it is based on public/portal/employee groups.

I suggest to check into existing fields and either use them either define a new one, computed / searchable in order to avoid having to hack the list view. Could you have a look ?

Thanks !

Reply:
Thanks for the hint.

After having a look, I was able to create a new computed field which contains the user type, it can be used directly in the tree view like a normal field.

You can check again.
Cheers!